python/tank/descriptor/io_descriptor/installed_config.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import os
+
+from .path import IODescriptorPath
+from .. import constants
+from ... import LogManager
+
+from ..errors import TankDescriptorError
+
+from tank_vendor import yaml
+
+log = LogManager.get_logger(__name__)
+
+
+class IODescriptorInstalledConfig(IODescriptorPath):
+    """
+    Installed descriptor type is a special type of io_descriptor that can only be used for installed
+    configurations and the InstalledConfigurationDescriptor. Its role is to differentiate
+    how the configuration folder is packaged inside a standard config descriptor and the installed
+    config descriptor. Functionally it operates exactly like a path descriptor, except that it
+    doesn't support copying itself.
+    """
```

Comment:
please add an example of descriptor syntax.

it seems we are violating othogonaly here - can you create a core or app descriptor which is of type installed config?

All other io descriptors are implementing a transport of a payload (app store, git etc) - this is implementing a type of content. This is different. Let's discuss if we can get rid of this class and move this up into the descriptor level, which is where the content specific stuff should be.